webapp/src/js/services/enketo.js

```diff
@@ -349,7 +356,28 @@ angular.module('inboxServices').service('Enketo',
         $(selector).on('valuechange.enketo', listener);
       }
     };
-                        
+
+    var registerValidationCompleteListener = function(selector) {
+      const listener = () => {
+        if ($('.invalid-required').length > 0) {
+          $(selector).find('.next-page').attr('disabled','disabled');
+        } else {
+          $(selector).find('.next-page').removeAttr('disabled');
+        }
+      };
+
+      if (listener) {
+        $(selector).on('valuechange.enketo', listener);
```

Comment:
According to the [changelog](https://github.com/enketo/enketo-core/blob/master/CHANGELOG.md#4180---2017-01-18) the `valuechange` event is passed the validity of the page, which _should_ be more accurate than checking for `invalid-required` classes. What do you think? I haven't tested that so YMMV...

Reply:
If we do this then we lose the pulling out as a new function suggestion you had above (since one of those calls doesn't have that event, so then the code is not similar enough to warrant it). Which would you prefer?

Reply:
I guess if we're relying on the class in one place we should rely on it everywhere for consistency. Pull it out and ignore the parameter.